test/integration/configurations/nighthawk_https_origin.yaml

```diff
@@ -1,40 +1,146 @@
+                        - "*"
+                http_filters:
+                  - name: test-server
+                    config:
+                      response_body_size: 10
+                      response_headers:
+                        - { header: { key: "x-nh", value: "1" } }
+                  - name: envoy.router
+                    config:
+                      dynamic_stats: false
+          tls_context:
+            common_tls_context:
+              tls_certificates:
+                - certificate_chain:
+                    inline_string: "-----BEGIN CERTIFICATE-----
+
+                      MIIEYTCCA0mgAwIBAgIJAILStmLgUUcVMA0GCSqGSIb3DQEBCwUAMHYxCzAJBgNV
+
+                      BAYTAlVTMRMwEQYDVQQIDApDYWxpZm9ybmlhMRYwFAYDVQQHDA1TYW4gRnJhbmNp
+
+                      c2NvMQ0wCwYDVQQKDARMeWZ0MRkwFwYDVQQLDBBMeWZ0IEVuZ2luZWVyaW5nMRAw
+
+                      DgYDVQQDDAdUZXN0IENBMB4XDTE4MTIxNzIwMTgwMFoXDTIwMTIxNjIwMTgwMFow
+
+                      gaYxCzAJBgNVBAYTAlVTMRMwEQYDVQQIDApDYWxpZm9ybmlhMRYwFAYDVQQHDA1T
+
+                      YW4gRnJhbmNpc2NvMQ0wCwYDVQQKDARMeWZ0MRkwFwYDVQQLDBBMeWZ0IEVuZ2lu
+
+                      ZWVyaW5nMRowGAYDVQQDDBFUZXN0IEJhY2tlbmQgVGVhbTEkMCIGCSqGSIb3DQEJ
+
+                      ARYVYmFja2VuZC10ZWFtQGx5ZnQuY29tMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8A
+
+                      MIIBCgKCAQEAuvPdQdmwZongPAgQho/Vipd3PZWrQ6BKxIb4l/RvqtVP321IUTLs
+
+                      4vVwpXoYJ+12L+XOO3jCInszs53tHjFpTI1GE8/sasmgR6LRr2krwSoVRHPqUoc9
+
+                      tzkDG1SzKP2TRTi1MTI3FO+TnLFahntO9Zstxhv1Epz5GZ/xQLE0/LLoRYzcynL/
+
+                      iflk18iL1KM8i0Hy4cKjclOaUdnh2nh753iJfxCSb5wJfx4FH1qverYHHT6FopYR
+
+                      V40Cg0yYXcYo8yNwrg+EBY8QAT2JOMDokXNKbZpmVKiBlh0QYMX6BBiW249v3sYl
+
+                      3Ve+fZvCkle3W0xP0xJw8PdX0NRbvGOrBQIDAQABo4HAMIG9MAwGA1UdEwEB/wQC
+
+                      MAAwCwYDVR0PBAQDAgXgMB0GA1UdJQQWMBQGCCsGAQUFBwMCBggrBgEFBQcDATBB
+
+                      BgNVHREEOjA4hh5zcGlmZmU6Ly9seWZ0LmNvbS9iYWNrZW5kLXRlYW2CCGx5ZnQu
+
+                      Y29tggx3d3cubHlmdC5jb20wHQYDVR0OBBYEFLHmMm0DV9jCHJSWVRwyPYpBw62r
+
+                      MB8GA1UdIwQYMBaAFBQz1vaSbPuePL++7GTMqLAMtk3kMA0GCSqGSIb3DQEBCwUA
+
+                      A4IBAQAwx3/M2o00W8GlQ3OT4y/hQGb5K2aytxx8QeSmJaaZTJbvaHhe0x3/fLgq
+
+                      uWrW3WEWFtwasilySjOrFOtB9UNmJmNOHSJD3Bslbv5htRaWnoFPCXdwZtVMdoTq
+
+                      IHIQqLoos/xj3kVD5sJSYySrveMeKaeUILTkb5ZubSivye1X2yiJLR7AtuwuiMio
+
+                      CdIOqhn6xJqYhT7z0IhdKpLNPk4w1tBZSKOXqzrXS4uoJgTC67hWslWWZ2VC6IvZ
+
+                      FmKuuGZamCCj6F1QF2IjMVM8evl84hEnN0ajdkA/QWnil9kcWvBm15Ho+oTvvJ7s
+
+                      M8MD3RDSq/90FSiME4vbyNEyTmj0
+
+                      -----END CERTIFICATE-----
+
+                      "
+                  private_key:
+                    inline_string: "-----BEGIN RSA PRIVATE KEY-----
+
+                      MIIEpAIBAAKCAQEAuvPdQdmwZongPAgQho/Vipd3PZWrQ6BKxIb4l/RvqtVP321I
```

Comment:
Generally not a fan of inlining these, since when the certs expire, it's hard to know how to regenerate them. All the test certs in the Envoy tree have associated scripts to rebuild them.

Reply:
So to make sure I get it right, you'd like to see https://github.com/envoyproxy/nighthawk/pull/337/commits/605e5ce39369a2ed93505fb8c0498849cf82bf70 back in, but then instead of parsing yaml we do a string based substitution?

Reply:
Yeah. My main issue with that PR was that it was doing sophisticated YAML parsing; I think that since YAML is a textual format, you can just substitute directly into it. If you use something like Python's string.Template format in the YAML, there's no need to even write a custom substituter.

Reply:
@htuch following our Slack discussion, I pushed 345972b